examples/scripts/erc721Deposit.ts

```diff
@@ -0,0 +1,112 @@
+
+// Example script
+const main = async () => {
+  let user;
+  let latestTokenId;
+
+  try {
+    // # 1 Create an instance of User (mnemonic is optional)
+    user = await UserFactory.create({
+      blockchainWsUrl: config.blockchainWsUrl,
+      clientApiUrl: config.clientApiUrl,
+      ethereumPrivateKey: config.ethereumPrivateKey,
+      nightfallMnemonic: config.nightfallMnemonic,
+    });
+    const contract = new web3.eth.Contract(
+      erc721AbiItem,
+      config.tokenContractAddress,
+    );
+
+    const userAddress = await web3.eth.accounts.privateKeyToAccount(
+      config.ethereumPrivateKey,
+    ).address;
+
+    //number of tokens owned by the address
+    const balance = await contract.methods.balanceOf(userAddress).call();
+
+    // create an NFT to be deposited to Nightfall
+    const tx = await contract.methods.awardItem(userAddress, "test");
+
+    const gas = await tx.estimateGas({
+      from: userAddress,
+    });
+
+    const gasPrice = await web3.eth.getGasPrice();
+    const data = tx.encodeABI();
+    const nonce = await web3.eth.getTransactionCount(userAddress);
+    const signedTx = await web3.eth.accounts.signTransaction(
+      {
+        to: config.tokenContractAddress,
+        data,
+        gas,
+        gasPrice,
+        nonce,
+      },
+      config.ethereumPrivateKey,
+    );
+
+    const receipt = await web3.eth.sendSignedTransaction(
+      signedTx.rawTransaction,
+      function (error, hash) {
+        if (!error) {
+          console.log(hash);
+        } else {
+          console.log(error);
+        }
+      },
+    );
+
+    // get the tokenId from the created NFT to deposit it to Nightfall
+    latestTokenId = await contract
+      .getPastEvents("Transfer", {
+        filter: {
+          _from: userAddress,
+        },
+        fromBlock: 0,
+      })
+      .then((events) => {
+        latestTokenId = events[events.length - 1].returnValues.tokenId;
+
+        console.log("The ID of the token to be deposited", latestTokenId);
+
+        return events[events.length - 1].returnValues.tokenId;
+      });
+
+    // # 4 Make ERC721 deposit
+    const tokenContractAddress = config.tokenContractAddress;
+    const tokenErcStandard = "ERC721";
+    const tokenId = latestTokenId;
+
+    // add only one to be required, not both value and tokenId
+    const txReceipts = await user.makeDeposit({
+      tokenContractAddress,
+      tokenErcStandard,
+      tokenId,
+    });
+
+    console.log("Transaction receipts", txReceipts);
+
+    // # 5 [OPTIONAL] You can check the transaction hash
+    console.log("Nightfall deposit tx hashes", user.nightfallDepositTxHashes);
+
+    // # 6 [OPTIONAL] You can check deposits that are not yet in Nightfall
+    const pendingDeposits = await user.checkPendingDeposits();
+    console.log("Pending balances", pendingDeposits);
+  } catch (error) {
+    console.log(error);
+    process.exit(1);
+  } finally {
+    user.close();
```

Comment:
@imagobea what do you think about this?
The process for the erc721 deposit isn't closing tho, just for this example. I need to close it with `ctrl c` after is finished.

Reply:
IMO we can ignore the closing issue. I'm pretty sure it will become obvious when we refactor the scripts. Right now the scripts perform quite a few operations, so we should break them down, perhaps into minting + deposit.

Pending deposits doesn't display pending ERC721 due to this line of code [e.value > 0](https://github.com/EYBlockchain/nightfall_3/blob/master/nightfall-client/src/services/commitment-storage.mjs#L370). We can check with dRo and Ze.